### PR TITLE
[Doc] update readme

### PR DESCRIPTION
**Pull request by rhdong**
- make readme clearer and simpler
- add contributors to readme
- add main features for making potential users get the points quickly
- fix some small errors
- add 0.4.0 version information in advance

### CODE REVIEW

**Review by MoFHeka — COMMENTED** [review]
LGTM

**Review by MoFHeka — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -5,66 +5,63 @@
...
-* [RFC: Embedding Variable](https://docs.google.com/document/d/1odez6-69YH-eFcp8rKndDHTNGxZgdFFRJufsW94_gl4)
+that makes TensorFlow more suitable for trainning models of **Search, Recommendations and Advertising** and 
+makes building, evaluating, and serving sophisticated recommenders models easy. 
+See approved TensorFlow RFC #[313](https://github.com/tensorflow/community/pull/313).
+Those contributions will be complementary to TensorFlow Core and TensorFlow Recommenders etc. 
+
+## Main Features
+
+- Make key-value data structure (dynamic embedding) trainable in TensorFlow
+- Get better recommendation effect compare to static embedding mechanism for no hash conflict
```
Comment (Lifann) [inline README.md:18]: compared to

Comment (Lifann) [inline README.md:18]: compared to

Comment (Lifann) [inline README.md:18]: with no hash conflicts

Comment (Lifann) [inline README.md:18]: with no hash conflicts

File: README.md [hunk 2]
```diff
@@ -5,66 +5,63 @@
...
+- Compatible with all native TensorFlow optimizers and initializers
+- Compatible with native TensorFlow CheckPoint and SavedModel format
+- Fully support train and inference recommeneders models on GPUs
+- Support [TF serving](https://github.com/tensorflow/serving) and [Triton Inference Server](https://github.com/triton-inference-server/server) as inference framework
+- Support variant Key-Value implements as dynamic embedding storage and easy to extend
+  - [cuckoohash_map](https://github.com/efficient/libcuckoo) (from Efficient Computing at Carnegie Mellon, on CPU)
+  - [nvhash](https://github.com/rapidsai/cudf) (from NVIDIA, on GPU)
+  - [Redis](https://github.com/redis/redis)
+- Support half synchronous training based on Horovod
```
Comment (Lifann) [inline README.md:27]: Maybe explain what half-synchronous training is

Comment (Lifann) [inline README.md:27]: Maybe explain what half-synchronous training is

File: README.md [hunk 3]
```diff
@@ -101,14 +98,14 @@ is compiled differently. A typical example of this would be `conda`-installed Te
...
 
-| TFRA | TensorFlow | Compiler  | CUDA | CUDNN | Compute Capability |
-|:----------------------- |:---- |:---------| :------------ | :---- | :------------ |
-| 0.3.0 | 2.5.1  | GCC 7.3.1 | 11.2 | 8.1 | 6.0, 6.1, 7.0, 7.5, 8.0, 8.6 |
-| 0.2.0 | 2.4.1  | GCC 7.3.1 | 11.0 | 8.0 | 6.0, 6.1, 7.0, 7.5, 8.0 |
-| 0.2.0 | 1.15.2  | GCC 7.3.1 | 10.0 | 7.6 | 6.0, 6.1, 7.0, 7.5 |
-| 0.1.0 | 2.4.1  | GCC 7.3.1 | - | - | - |
+| TFRA  | TensorFlow | Compiler  | CUDA | CUDNN | Compute Capability |
+|:------|:---- |:---------| :------------ | :---- | :------------ |
+| 0.3   | 2.5.1  | GCC 7.3.1 | 11.2 | 8.1 | 6.0, 6.1, 7.0, 7.5, 8.0, 8.6 |
```
Comment (Lifann) [inline README.md:105]: Will the patch version number be ignored like: `pip install tensorflow-recommenders-addons==0.3`?

Comment (Lifann) [inline README.md:105]: Will the patch version number be ignored like: `pip install tensorflow-recommenders-addons==0.3`?

File: README.md [hunk 4]
```diff
@@ -136,7 +133,7 @@ Please install a TensorFlow on your compiling machine, The compiler needs to kno
 its headers according to the installed TensorFlow. 
 
 ```
-export TF_VERSION=["2.5.1", "2.4.1", "1.15.2"]
+export TF_VERSION="2.5.1"  # "2.5.1", "2.4.1", "1.15.2" are valid.
```
Comment (Lifann) [inline README.md:136]: Maybe are verified is more broadly fit for users with different TF versions, when compiling from source.

Comment (Lifann) [inline README.md:136]: Maybe are verified is more broadly fit for users with different TF versions, when compiling from source.

File: README.md (hunk 1 above)
Comment (rhdong) [reply README.md:18]: accept

File: README.md (hunk 4 above)
Comment (rhdong) [reply README.md:136]: export TF_VERSION="2.5.1"  # "2.7.0", "2.5.1" are well tested.

File: README.md (hunk 3 above)
Comment (rhdong) [reply README.md:105]: add .0 back

**Review by Lifann — APPROVED** [review]
LGTM

**Pull request merged**